### ISSUE TITLE
Floating point exception when running torch.nn.functional.conv2d

### ISSUE THREAD

**Issue by nimashiri** [pytorch/pytorch#91558]
### 🐛 Describe the bug

If torch.nn.functional.conv2d given zero element for last parameter, it results in floating point exception on pytorch 1.8.0 and 1.12.0. 

```
import torch
import numpy as np
arg_1_tensor = torch.rand([4, 8, 224, 224], dtype=torch.float32)
arg_1 = arg_1_tensor.clone()
arg_2_tensor = torch.rand([4, 2, 3, 3], dtype=torch.float32)
arg_2 = arg_2_tensor.clone()
arg_3_tensor = torch.rand([4], dtype=torch.float32)
arg_3 = arg_3_tensor.clone()
arg_4_0 = 2
arg_4_1 = 2
arg_4 = [arg_4_0,arg_4_1,]
arg_5_0 = 1
arg_5_1 = 1
arg_5 = [arg_5_0,arg_5_1,]
arg_6_0 = 1
arg_6_1 = 1
arg_6 = [arg_6_0,arg_6_1,]
arg_7 = 0
try:
  res = torch.nn.functional.conv2d(arg_1,arg_2,arg_3,arg_4,arg_5,arg_6,arg_7,)
except Exception as e:
  print("Error:"+str(e))
```

### Versions

Collecting environment information...
PyTorch version: 1.12.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.22.6
Libc version: glibc-2.27

Python version: 3.8.16 (default, Dec  7 2022, 01:12:13)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.10.133+-x86_64-with-glibc2.27
Is CUDA available: False
CUDA runtime version: 11.2.152
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.6
[pip3] torch==1.12.0
[pip3] torchaudio==0.13.0+cu116
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.14.0
[pip3] torchvision==0.14.0+cu116
[conda] Could not collect

**Comment by GuichardVictor** [comment]
Hi,

The last parameter of `torch.nn.functional.conv2d` is according to the documentation:
> groups – split input into groups, \text{in\_channels}in_channels should be divisible by the number of groups. Default: 1

A `floating point exception` occurs for impossible operations, such as dividing by zero, which is what happens when `groups=0`.

So I would say this is expected.

**Comment by nimashiri** [comment]
> Hi,
> 
> The last parameter of `torch.nn.functional.conv2d` is according to the documentation:
> 
> > groups – split input into groups, \text{in_channels}in_channels should be divisible by the number of groups. Default: 1
> 
> A `floating point exception` occurs for impossible operations, such as dividing by zero, which is what happens when `groups=0`.
> 
> So I would say this is somewhat expected.

Hi, so this is not considered to be a bug? I think atleast a proper exception handling should be done to guard against denial of service via crash.

**Comment by GuichardVictor** [comment]
It could be considered a bug as it crashes a google colab instance. However this was fixed in torch 1.13 with this given message:
> non-positive groups is not supported

**Comment by nimashiri** [comment]
> It could be considered a bug as it crashes a google colab instance. However this was fixed in torch 1.13 with this given message:
> 
> > non-positive groups is not supported

ok thanks.

**Comment by soumith** [comment]
closing as @GuichardVictor confirmed that this was fixed in 1.13

**Comment by nimashiri** [comment]
Also this one: 
```
import torch
import numpy as np
arg_1_tensor = torch.rand([1, 3, 224, 224], dtype=torch.float32)
arg_1 = arg_1_tensor.clone()
arg_2_tensor = torch.rand([64, 3, 7, 7], dtype=torch.float32)
arg_2 = arg_2_tensor.clone()
arg_3_tensor = torch.rand([64], dtype=torch.float32)
arg_3 = arg_3_tensor.clone()
arg_4_0 = 2
arg_4_1 = 2
arg_4 = [arg_4_0,arg_4_1,]
arg_5_0 = 3
arg_5_1 = 3
arg_5 = [arg_5_0,arg_5_1,]
arg_6_0 = 44
arg_6_1 = 13
arg_6 = [arg_6_0,arg_6_1,]
arg_7 = 0
try:
  res = torch.nn.functional.conv2d(arg_1,arg_2,arg_3,arg_4,arg_5,arg_6,arg_7,)
except Exception as e:
  print("Error:"+str(e))

```
